model/user/user.go

```diff
@@ -166,6 +168,53 @@ func (u *DBUser) UpdateParsleySettings(settings parsley.Settings) error {
 	return nil
 }
 
+// CheckAndUpdateSchedulingLimit checks if the number of tasks to be activated by the user is allowed given
+// the global per-user hourly task scheduling limit, and updates relevant timestamp and counter info used
+// to track the user's hourly scheduling usage. The numTasksActivated parameter being negative signifies
+// the user is deactivating tasks, which frees up space in their scheduling limit.
+func (u *DBUser) CheckAndUpdateSchedulingLimit(settings *evergreen.Settings, numTasksActivated int) error {
```

Comment:
could we re-name numTasksActivated to something like numTasksChanged or something like that so it's more intuitive that it could be negative?

Reply:
Good point, changed.